src/api/posts/src/storage.js

```diff
@@ -22,6 +22,26 @@ const createInvalidFeedKey = (id) => createFeedKey(id).concat(invalidSuffix);
 // "NirlSYranl" to "t:feed:NirlSYranl:delayed"
 const createDelayedFeedKey = (id) => createFeedKey(id).concat(delayedSuffix);
 
+const getInvalidKeys = async () => {
+  let keysArray = [];
+  const stream = redis.scanStream({
+    match: 't:feed:*:invalid',
+  });
+  return new Promise((res, rej) => {
+    stream.on('data', async (keys) => {
+      keysArray = keysArray.concat(keys);
+    });
+    stream.on('end', () => {
+      //remove duplicate keys
+      let filterArray = [...new Set(keysArray)];
+      res(filterArray);
+    });
```

Comment:
Let's add something deal with the error case:

```js
stream.on('error', (err) => {
  logger.error({ err }, 'Error while scanning redis keys');
  reject([]);
});
```

Reply:
This is great recommendation, thank you